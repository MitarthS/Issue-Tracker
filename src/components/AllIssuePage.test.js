import React from 'react';
import {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AllIssuePage from './AllIssuesPage';

describe('When no props is passed to AllIssuePage ', () => {
    it('renders correctly', () => {
        let wrapper = shallow(<AllIssuePage/>);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('When issues array is passed to AllIssuePage', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            issues: [
                {
                    "id":1,
                    "description": "Raj",
                    "serverity": "Critical",
                    "status": "Open"                    
                },
                {
                    "id":2,
                    "description": "Ramesh",
                    "serverity": "Critical",
                    "status": "Open"
                }
            ]
        }
        wrapper = shallow(<AllIssuePage {...props}/>);
    });

    it('renders correctly when passed the  props', () => {
        expect(wrapper).toMatchSnapshot();
    });

});

describe('When issues array passed to AllIssuePage is null', () => {
    let wrapper;
    let props;

    beforeEach(() => {
        props = {
            issues: null
        }
        wrapper = shallow(<AllIssuePage {...props}/>);
    });

    it('should not crash', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
