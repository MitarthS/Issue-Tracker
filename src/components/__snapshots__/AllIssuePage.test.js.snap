// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When issues array is passed to AllIssuePage renders correctly when passed the  props 1`] = `ShallowWrapper {}`;

exports[`When issues array passed to AllIssuePage is null should not crash 1`] = `ShallowWrapper {}`;

exports[`When no props is passed to AllIssuePage  renders correctly 1`] = `ShallowWrapper {}`;
